sympy/physics/mechanics/joint.py

```diff
@@ -474,6 +474,55 @@ def _locate_joint_frame(self, body, interframe):
         body.masscenter.set_vel(interframe, 0)  # Fixate interframe to body
         return interframe
 
+    def _fill_coordinate_list(self, coordinates, n_coords, label='q', offset=0,
+                              number_single=False):
+        """Helper method for _generate_coordinates and _generate_speeds.
+
+        Parameters
+        ==========
+
+        coordinates : iterable
+            Iterable of coordinates or speeds that have been provided.
+        n_coords : Integer
+            Number of coordinates that should be returned.
+        label : String, optional
+            Coordinate type either 'q' (coordinates) or 'u' (speeds). The
+            Default is 'q'.
+        offset : Integer
+            Count offset when creating new dynamicsymbols. The default is 0.
+        number_single : Boolean
+            Boolean whether if n_coords == 1, number should still be used. The
+            default is False.
+
+        """
+
+        def create_symbol(number):
+            if n_coords == 1 and not number_single:
+                return dynamicsymbols(f'{label}_{self.name}')
+            return dynamicsymbols(f'{label}{number}_{self.name}')
+
+        name = 'generalized coordinate' if label == 'q' else 'generalized speed'
+        generated_coordinates = []
+        if coordinates is None:
+            coordinates = []
+        elif isinstance(coordinates, (AppliedUndef, Derivative)):
+            coordinates = [coordinates]
+        # Supports more iterables, also Matrix
+        for i, coord in enumerate(coordinates):
+            if coord is None:
+                generated_coordinates.append(create_symbol(i + offset))
+            elif isinstance(coord, (AppliedUndef, Derivative)):
+                generated_coordinates.append(coord)
+            else:
+                raise TypeError(f'The {name} {coord} should have been a '
+                                f'dynamicsymbol.')
+        for i in range(len(coordinates) + offset, n_coords + offset):
+            generated_coordinates.append(create_symbol(i))
+        if len(generated_coordinates) != n_coords:
+            raise ValueError(f'{len(coordinates)} {name}s have been provided. '
+                             f'The maximum number of {name}s is {n_coords}.')
+        return Matrix(generated_coordinates)
```

Comment:
I didn't notice any tests for this method. Could you add a few regression tests for it?

Reply:
Forgot those, thanks for noting. They've been added.